sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOReadTest.java

```diff
@@ -404,6 +407,65 @@ public void processElement(ProcessContext c) throws Exception {
     p.run();
   }
 
+  @Test
+  public void testReadTableWithSchema() throws IOException, InterruptedException {
+    // setup
+    Table someTable = new Table();
+    someTable.setSchema(
+        new TableSchema()
+            .setFields(
+                ImmutableList.of(
+                    new TableFieldSchema().setName("name").setType("STRING"),
+                    new TableFieldSchema().setName("number").setType("INTEGER"))));
```

Comment:
test should include a nested row as well.

Reply:
I discovered a bug in the fake JobService while trying to do this. It'll require some non trivial amount of code changes so I thought it might be better to fix that in a separate PR from this. What do you think?

Issue link: https://issues.apache.org/jira/browse/BEAM-7506

Reply:
Delaying to a separate PR sounds fine